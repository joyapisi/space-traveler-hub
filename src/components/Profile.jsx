import { useSelector } from 'react-redux';

const Profile = () => {
  const missions = useSelector((store) => store.missions.missions);
  const rocket = useSelector((store) => store.rocket.rocket);

  const myMissions = missions.filter((mission) => mission.reserved);
  const myRockets = rocket.filter((rocket) => rocket.reserved);

  return (
    <div className="main-profile-container">
      <div className="mission-profile-container">
        <h2 className="mission-title">my missions</h2>
        <table className="mission-item">
          <tbody>
            <>
              {myMissions.map((mission) => (
                <tr key={mission.mission_id}>
                  <td>{mission.mission_name}</td>
                </tr>
              ))}
            </>
          </tbody>
        </table>
      </div>
      <div className="rocket-profile-container">
        <h2 className="rocket-title">my rockets</h2>
        <table className="rocket-item">
          <tbody>
            <>
              {myRockets.map((rocket) => (
                <tr key={rocket.id}>
                  <td>{rocket.name}</td>
                </tr>
              ))}
            </>
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default Profile;
