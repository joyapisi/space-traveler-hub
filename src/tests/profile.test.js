import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import Store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Profile from '../components/Profile';

describe('Profile', () => {
  test('Render Profile Component', () => {
    const tree = TestRenderer.create(
      <Provider store={Store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
