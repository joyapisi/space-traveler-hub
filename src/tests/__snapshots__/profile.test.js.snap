// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Render Profile Component 1`] = `
<div
  className="main-profile-container"
>
  <div
    className="mission-profile-container"
  >
    <h2
      className="mission-title"
    >
      my missions
    </h2>
    <table
      className="mission-item"
    >
      <tbody />
    </table>
  </div>
  <div
    className="rocket-profile-container"
  >
    <h2
      className="rocket-title"
    >
      my rockets
    </h2>
    <table
      className="rocket-item"
    >
      <tbody />
    </table>
  </div>
</div>
`;
